packages/gitbook/src/middleware.ts

```diff
@@ -172,11 +172,14 @@ export async function middleware(request: NextRequest) {
             // Start fetching everything as soon as possible, but do not block the middleware on it
             // the cache will handle concurrent calls
             await waitUntil(
-                getSpaceContentData({
-                    spaceId: resolved.space,
-                    changeRequestId: resolved.changeRequest,
-                    revisionId: resolved.revision,
-                }),
+                getSpaceContentData(
+                    {
+                        spaceId: resolved.space,
+                        changeRequestId: resolved.changeRequest,
+                        revisionId: resolved.revision,
+                    },
+                    'site' in resolved ? resolved.shareKey : undefined,
```

Comment:
I'm wondering if we shouldn't have a helper for that, given how many times it's done..!

Reply:
I agree but all of this will be refactored and cleaned up soon when sites are 100% GA